### PR TITLE
Chore/speed up inprocess

### PR DESCRIPTION
**Pull request by varsis**
- Speed up InProcess driver tests

### CODE REVIEW

File: src/driver/AsyncJobs.ts [hunk 1]
```diff
@@ -1,19 +1,36 @@
...
+    pushJobs(jobs: Array<Promise<any>>): void {
+        this.jobs.concat(jobs)
+    }
+
     pushJob(job: Promise<any>): void {
         this.jobs.push(job)
     }
 
     async jobsComplete(): Promise<void> {
-        // More jobs might be added by each job, so we can't just await Promise.all() here.
```
Comment (hughgrigg) [inline src/driver/AsyncJobs.ts]: Why did you remove the comment? I think it's quite helpful to explain that here.

File: src/driver/AsyncJobs.ts [hunk 2]
```diff
@@ -1,19 +1,36 @@
...
+                }
+            })
+
+            await Promise.all(itemsToResolve.map((x) => x.promise))
+
+            // Assumption is that the items are at the start
+            this.jobs.splice(
+                0,
+                Math.max(...itemsToResolve.map((x) => x.index)) + 1,
+            )
```
Comment (hughgrigg) [inline src/driver/AsyncJobs.ts]: I don't follow what this change does. Can you explain it?

**Comment by hughgrigg** [comment]
@varsis Can you get some metrics on what kind of difference this makes to speed?

**Comment by varsis** [comment]
```
   console.info src/util/performance.ts:5
      triggerChangeEvents: before: Execution time (hr): 0s 0.262799ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before, makeChange: Execution time (hr): 0s 0.278171ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: after: Execution time (hr): 0s 0.145049ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: pushJobs: Execution time (hr): 0s 0.129355ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before: Execution time (hr): 0s 0.015528ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before, makeChange: Execution time (hr): 0s 0.0372ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: after: Execution time (hr): 0s 0.084483ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: pushJobs: Execution time (hr): 0s 0.003641ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before: Execution time (hr): 0s 0.066348ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before, makeChange: Execution time (hr): 0s 0.048627ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: after: Execution time (hr): 0s 0.197233ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: pushJobs: Execution time (hr): 0s 0.17623ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before: Execution time (hr): 0s 0.012118ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before, makeChange: Execution time (hr): 0s 0.040871ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: after: Execution time (hr): 0s 0.164105ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: pushJobs: Execution time (hr): 0s 0.181158ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before: Execution time (hr): 0s 0.012329ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before, makeChange: Execution time (hr): 0s 0.049577ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: after: Execution time (hr): 0s 0.067098ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: pushJobs: Execution time (hr): 0s 0.005767ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before: Execution time (hr): 0s 0.241388ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before, makeChange: Execution time (hr): 0s 0.152689ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: after: Execution time (hr): 0s 0.128317ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: pushJobs: Execution time (hr): 0s 0.059168ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before: Execution time (hr): 0s 0.010715ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before, makeChange: Execution time (hr): 0s 0.03125ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: after: Execution time (hr): 0s 0.079476ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: pushJobs: Execution time (hr): 0s 0.034636ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before: Execution time (hr): 0s 0.010419ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before, makeChange: Execution time (hr): 0s 0.025906ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: after: Execution time (hr): 0s 0.067506ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: pushJobs: Execution time (hr): 0s 0.02849ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before: Execution time (hr): 0s 0.049262ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before, makeChange: Execution time (hr): 0s 0.02553ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: after: Execution time (hr): 0s 0.093949ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: pushJobs: Execution time (hr): 0s 0.023988ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before: Execution time (hr): 0s 0.073207ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before, makeChange: Execution time (hr): 0s 0.025323ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: after: Execution time (hr): 0s 0.76265ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: pushJobs: Execution time (hr): 0s 0.020878ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before: Execution time (hr): 0s 0.011072ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before, makeChange: Execution time (hr): 0s 0.031391ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: after: Execution time (hr): 0s 0.075776ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: pushJobs: Execution time (hr): 0s 0.029026ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before: Execution time (hr): 0s 5.82065ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before, makeChange: Execution time (hr): 0s 0.038394ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: after: Execution time (hr): 0s 0.11516ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: pushJobs: Execution time (hr): 0s 0.020071ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before: Execution time (hr): 0s 0.109539ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before, makeChange: Execution time (hr): 0s 0.028867ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: after: Execution time (hr): 0s 0.16453ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: pushJobs: Execution time (hr): 0s 0.02592ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before: Execution time (hr): 0s 0.882366ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before, makeChange: Execution time (hr): 0s 0.040165ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: after: Execution time (hr): 0s 0.233355ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: pushJobs: Execution time (hr): 0s 0.027767ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before: Execution time (hr): 0s 0.195605ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before, makeChange: Execution time (hr): 0s 0.037511ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: after: Execution time (hr): 0s 0.29658ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: pushJobs: Execution time (hr): 0s 0.020695ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before: Execution time (hr): 0s 0.223456ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before, makeChange: Execution time (hr): 0s 0.027313ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: after: Execution time (hr): 0s 0.272274ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: pushJobs: Execution time (hr): 0s 0.020617ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before: Execution time (hr): 0s 0.008613ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before, makeChange: Execution time (hr): 0s 0.063419ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: after: Execution time (hr): 0s 0.088644ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: pushJobs: Execution time (hr): 0s 0.027006ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before: Execution time (hr): 0s 0.046102ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before, makeChange: Execution time (hr): 0s 0.020465ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: after: Execution time (hr): 0s 0.094019ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: pushJobs: Execution time (hr): 0s 0.126732ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before: Execution time (hr): 0s 0.094191ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before, makeChange: Execution time (hr): 0s 0.022321ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: after: Execution time (hr): 0s 0.129043ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: pushJobs: Execution time (hr): 0s 0.021851ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before: Execution time (hr): 0s 0.008239ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before, makeChange: Execution time (hr): 0s 0.01604ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: after: Execution time (hr): 0s 0.050179ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: pushJobs: Execution time (hr): 0s 0.019253ms

```

```
console.info src/util/performance.ts:5
      triggerChangeEvents: before: Execution time (hr): 0s 0.061059ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before, makeChange: Execution time (hr): 0s 0.17119ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: after: Execution time (hr): 0s 0.089599ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: after, makeChange: Execution time (hr): 0s 0.330287ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: pushJobs: Execution time (hr): 0s 0.096337ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before: Execution time (hr): 0s 0.018217ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: before, makeChange: Execution time (hr): 0s 0.024623ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: after: Execution time (hr): 0s 0.061655ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: after, makeChange: Execution time (hr): 0s 0.047007ms
    console.info src/util/performance.ts:5
      triggerChangeEvents: pushJobs: Execution time (hr): 0s 0.003565ms
```
